### PR TITLE
NIAD-588 Map ITK -> QuestionnaireResponse, Questionare

### PR DESCRIPTION
**Pull request by Broom94**
Questionnaire, Questionnaire mappers complete with dateutil amendments. Two extra util helpers created structuredBodyutil and Pathwayutil. Bundle update to add question and questionnaires. Tests created and amended.

### CODE REVIEW

**Review by dawidskainos — CHANGES_REQUESTED** [review]
Questionnaire and QuestionnaireResponse resources are missing fullUrls and are not included in List resource.

**Review by dawidskainos — DISMISSED** [review]

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/QuestionnaireMapper.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            .setExperimental(false)
+            .addSubjectType("Patient")
+            .setDate(latestDate)
+            .setPublisher(publisher)
+            .setLastReviewDate(latestDate)
+            .setJurisdiction(Collections.singletonList(
+                new CodeableConcept().setText(
+                    getCountry(pathwaysCase))))
+            .addContact(
+            new ContactDetail().addTelecom(
```
Comment (saddiqs1) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/QuestionnaireMapper.java:50]: lines 50-53 need indenting?

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/QuestionnaireMapper.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                new ContactPoint().setValue(
+                    getContactNumber(pathwaysCase)
+                )))
+            .addItem(getItem(triageLine.getQuestion(), getCaseID(pathwaysCase)));
+
+        return questionnaire;
+    }
+
+    private String getPublisher(User user) {
+        String value = "";
```
Comment (saddiqs1) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/QuestionnaireMapper.java:60]: `StringUtils.EMPTY` is preferred to `""`

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/QuestionnaireMapper.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        if (user.getId() != null) {
+            value += String.format("User ID: '%s' ", user.getId());
+        }
+        if (user.getName() != null) {
+            value += String.format("User name: '%s' ", user.getName());
+        }
+        if (user.getSkillSet() != null) {
+            value += String.format("User skill set: '%s'", user.getSkillSet());
+        }
+        if (value.equals("")) {
```
Comment (saddiqs1) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/QuestionnaireMapper.java:70]: `value.isBlank()` is preferable?

**Review by dawidskainos — APPROVED** [review]

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/util/StructuredBodyUtil.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+        return Optional.ofNullable(clinicalDocument)
+            .map(POCDMT000002UK01ClinicalDocument1::getComponent)
+            .map(POCDMT000002UK01Component2::getStructuredBody)
+            .orElse(null);
+    }
+
+    public Predicate<? super POCDMT000002UK01Entry> entryHasTemplate(String template) {
+        return entry -> entry.isSetContentId()
+            && entry.getContentId().getRoot().equals("2.16.840.1.113883.2.1.3.2.4.18.16")
```
Comment (saddiqs1) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/util/StructuredBodyUtil.java:30]: Extract string here into constant

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/QuestionnaireResponseMapper.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        return questionnaireResponse;
+    }
+
+    private QuestionnaireResponseItemComponent getItem(Question question) {
+        QuestionnaireResponseItemComponent item = new QuestionnaireResponseItemComponent();
+        QuestionnaireResponseItemAnswerComponent answer = new QuestionnaireResponseItemAnswerComponent();
+        StringType correctAnswerText = new StringType();
+
+        if (question.getQuestionId() == null || question.getQuestionId().isBlank()) {
+            item.setLinkId("N/A");
```
Comment (saddiqs1) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/QuestionnaireResponseMapper.java:62]: Few references of `N/A` throughout multiple files, maybe make into a constant somewhere?

**Review by saddiqs1 — APPROVED** [review]

**Review by dawidskainos — APPROVED** [review]

**Pull request merged**